pkg/api/defaults.go

```diff
@@ -376,6 +376,12 @@ func (p *Properties) setVMSSDefaultsForMasters() {
 		p.OrchestratorProfile.KubernetesConfig.LoadBalancerSku = "Standard"
 		p.OrchestratorProfile.KubernetesConfig.ExcludeMasterFromStandardLB = helpers.PointerToBool(DefaultExcludeMasterFromStandardLB)
 	}
+	k8sConfig := p.OrchestratorProfile.KubernetesConfig
+	// We use user assigned identity (a known default one) for vmss masters.
+	if k8sConfig != nil {
+		k8sConfig.UseManagedIdentity = true
+		k8sConfig.UserAssignedID = DefaultUserAssignedID
```

Comment:
The logic here should be: if it's vmss master and if user wants to use managed identity, then we default them to `UserAssignedIDEnabled` by assigning a default user assigned id. 

```go
if k8sConfig != nil && k8sConfig.UseManagedIdentity == true {
		k8sConfig.UserAssignedID = DefaultUserAssignedID
```

Reply:
Can we a add unit test to test this? Sorry for not asking earlier :( , but defaults is a pretty critical area in acs_engine and we need all the coverage we can get .